accepted/2021/statics-in-interfaces/README.md

```diff
@@ -8,6 +8,793 @@ C# is looking at enabling static abstract members in interfaces (https://github.
+
+```csharp
+namespace System
+{
+    public struct Byte : IUnsignedBinaryInteger<byte> { }
+
+    public struct Char : IUnsignedBinaryInteger<char> { }
+
+    public struct DateTime
+        : IAddable<DateTime, TimeSpan>,
+          IComparableOperators<DateTime, DateTime> // implies IEquatableOperators<DateTime, DateTime>
+    {
+        // MinValue and MaxValue should likely come from a base interface
+
+        static DateTime MinValue { get; }
+
+        static DateTime MaxValue { get; }
+    }
+
+    public struct DateTimeOffset
+        : IAddable<DateTimeOffset, DateTimeOffset>,
+          IAddable<DateTimeOffset, TimeSpan>,
+          IComparableOperators<DateTimeOffset, DateTimeOffset> // implies IEquatableOperators<DateTimeOffset, DateTimeOffset>
+    {
+        // MinValue and MaxValue should likely come from a base interface
+
+        static DateTimeOffset MinValue { get; }
+
+        static DateTimeOffset MaxValue { get; }
+
+        // TODO: DateTimeOffset defines an implicit conversion to DateTime, should that be modeled in the interfaces?
+    }
+
+    public struct Decimal
+        : ISignedNumber<decimal>
+    {
+        // Decimal defines a few additional operations like Ceiling, Floor, Round, and Truncate
+        // The rest of the IEEE operations are missing.
+
+        // Decimal and some other types likewise define "friendly" names for operators, should we expose them?
+    }
+
+    public struct Double : IBinaryFloatingPoint<double> { }
+
+    public struct Guid
+        : IComparableOperators<Guid, Guid> // implies IEquatableOperators<Guid, Guid>
+    {
+    }
+
+    public struct Half : IBinaryFloatingPoint<Half> { }
+
+    public struct Int16 : ISignedBinaryInteger<short> { }
+
+    public struct Int32 : ISignedBinaryInteger<int> { }
+
+    public struct Int64 : ISignedBinaryInteger<long> { }
+
+    public struct IntPtr : ISignedBinaryInteger<nint> { /* Size */ }
+
+    public struct SByte : ISignedBinaryInteger<sbyte> { }
+
+    public struct Single : IBinaryFloatingPoint<float> { }
+
+    public struct TimeSpan
+        : IAddable<TimeSpan, TimeSpan>,
+          IComparableOperators<TimeSpan, TimeSpan>, // implies IEquatableOperators<TimeSpan, TimeSpan>
+          IDivisable<TimeSpan, TimeSpan>,
+          IDivisable<TimeSpan, double>,
+          IMultipliable<TimeSpan, TimeSpan>,
+          IMultipliable<TimeSpan, double>,
+          INegatable<TimeSpan>
+    {
+    }
+
+    public struct UInt16 : IUnsignedBinaryInteger<ushort> { }
+
+    public struct UInt32 : IUnsignedBinaryInteger<uint> { }
+
+    public struct UInt64 : IUnsignedBinaryInteger<ulong> { }
+
+    public struct UIntPtr : IUnsignedBinaryInteger<nuint> { }
+}
+```
+
+### Pending Concepts
+
+There are several types which may benefit from some interface support. These include, but aren't limited to:
+* System.Array
+* System.DateOnly
+* System.Enum
+* System.Index
+* System.Range
+* System.TimeOnly
+* System.Tuple
+* System.ValueTuple
```

Comment:
ValueTuple is instantiated a lot. Adding more interfaces to it is very problematic performance-wise.